metamist/audit/adapters/storage_client.py

```diff
@@ -0,0 +1,234 @@
+
+    def check_blob_exists(self, bucket_name: str, blob_name: str) -> bool:
+        """
+        Check if a blob exists.
+
+        Args:
+            bucket_name: Name of the bucket
+            blob_name: Name of the blob
+
+        Returns:
+            True if blob exists, False otherwise
+        """
+        bucket_name = bucket_name.removeprefix('gs://').removesuffix('/')
+        bucket = self.client.bucket(bucket_name, user_project=self.project)
+        blob = bucket.blob(blob_name)
+        return blob.exists()
+
+    def find_files_in_prefixes(
+        self, bucket_prefixes: Dict[str, List[str]], file_extensions: tuple[str, ...]
+    ) -> Dict[str, List[str]]:
+        """
+        Find files with specific extensions in multiple bucket/prefix combinations.
+
+        Args:
+            bucket_prefixes: Dictionary of bucket names to list of prefixes
+            file_extensions: Tuple of file extensions to find
+
+        Returns:
+            Dictionary of prefixes to list of file paths found
+        """
+        files_by_prefix: Dict[str, List[str]] = {}
+
+        # Skip hail table/matrix table/vds paths
+        hail_extensions = ['.ht', '.mt', '.vds']
+
+        for bucket_name, prefixes in bucket_prefixes.items():
+            bucket = self.client.bucket(bucket_name, user_project=self.project)
+
+            for prefix in prefixes:
+                # Skip if prefix contains hail extensions
+                if any(ext in prefix for ext in hail_extensions):
+                    continue
+
+                found_files = []
+                for blob in self.client.list_blobs(bucket, prefix=f'{prefix}/'):
+                    if blob.name.endswith(file_extensions):
+                        found_files.append(f'gs://{bucket_name}/{blob.name}')
+
+                files_by_prefix[prefix] = found_files
+
+        return files_by_prefix
+
+    def extract_bucket_and_prefixes(self, paths: List[str]) -> Dict[str, List[str]]:
+        """
+        Extract unique bucket and prefix combinations from a list of paths.
+
+        Args:
+            paths: List of GCS paths
+
+        Returns:
+            Dictionary of bucket names to list of unique prefixes
+        """
+        buckets_prefixes: Dict[str, List[str]] = defaultdict(list)
+
+        for path_str in paths:
+            try:
+                path = to_path(path_str)
+            except ValueError:
+                continue
+
+            bucket = path.bucket
+            # Get prefix (everything except the filename)
+            prefix = '/'.join(path.parts[2:-1])  # Skip 'gs://', bucket, and filename
+
+            if prefix and prefix not in buckets_prefixes[bucket]:
+                buckets_prefixes[bucket].append(prefix)
+
+        return dict(buckets_prefixes)
+
+    def delete_blobs(
+        self,
+        paths: List[str],
+    ):
+        """
+        Delete blobs from Google Cloud Storage.
+        """
+        for path in paths:
+            self.delete_blob(path)
```

Comment:
IDK how much real time would be saved here, but if audits are per-bucket, this multiple delete could be more efficient by creating a single instance of `self.client.get_bucket(bucket_name)` (in self.delete_blob) and rolling multiple deletes on the same instance. Currently this creates a new client.bucket with each file.